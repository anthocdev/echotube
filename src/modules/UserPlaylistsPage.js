import React from "react";
import { connect } from "react-redux";
import * as PlaylistActions from "../actions/PlaylistActions";
import Playlists from "./containers/UserPlaylists";
import CreatePlaylistDialog from "./containers/Forms/PlaylistManagement/CreatePlaylistDialog";

/* Page for displaying playlists of signed in user */
class UserPlaylistsPage extends React.Component {
  componentDidMount() {
    this.props.getUserPlaylists();
  }

  render() {
    if (!this.props.userPlaylist.Playlists.length) {
      return <div>--No Playlists--</div>;
    }
    return (
      <div>
        <h1> Playlists </h1>
        <Playlists playlists={this.props.userPlaylist.Playlists} />
        <CreatePlaylistDialog />
      </div>
    );
  }
}

function mapStateToProps(state, props) {
  return {
    userPlaylist: state.userPlaylist,
  };
}

export default connect(mapStateToProps, PlaylistActions)(UserPlaylistsPage);
