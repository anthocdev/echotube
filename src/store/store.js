import { applyMiddleware, createStore, compose } from "redux";
import { createLogger } from "redux-logger";
import { default as thunk } from "redux-thunk";
import { createPromise } from "redux-promise-middleware";
import axios from "axios";

import reducer from "../reducers";
//Pulling JWT Token from local storage
const jwttoken = localStorage.getItem("JWT_TOKEN");
//Setting JWT Token as common header for axios
axios.defaults.headers.common["Authorization"] = jwttoken;
//Create store with middleware included
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; //Adding redux dev tool compatibility
const middleware = applyMiddleware(createPromise(), thunk, createLogger()); //Adding thunk and console logging for testing as middleware

export default createStore(
  reducer,
  {
    //Initializing auth state if JWT token is available
    auth: {
      jwtToken: jwttoken,
      isAuth: jwttoken ? true : false
    }
  },
  composeEnhancers(middleware)
);
