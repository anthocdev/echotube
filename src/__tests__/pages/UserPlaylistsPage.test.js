import React from "react";
import UserPlaylistsPage from "../../modules/UserPlaylistsPage";
import { Provider } from "react-redux";
import store from "../../store/store";
import { create, act } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("User Playlists Page Tests", () => {
  let root;

  it("Renders Without Crashing", async () => {
    act(() => {
      root = create(
        <Provider store={store}>
          <UserPlaylistsPage />
        </Provider>
      );
    });
  });

  test("Renders Matches Snapshot", async () => {
    act(() => {
      root = create(
        <BrowserRouter>
          <Provider store={store}>
            <UserPlaylistsPage />
          </Provider>
        </BrowserRouter>
      );
    });
    expect(root).toMatchSnapshot();
    console.log(root.toJSON());
  });
});
