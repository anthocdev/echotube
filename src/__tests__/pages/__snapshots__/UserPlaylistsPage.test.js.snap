// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Playlists Page Tests Renders Matches Snapshot 1`] = `
<div
  className="playlistheader"
>
  --No Playlists-- 
  <div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.3)",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Create a New Playlist
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
